btsmarthub_devicelist/__init__.py

```diff
@@ -0,0 +1,342 @@
+import requests
+from urllib.parse import quote
+import json
+import math
+import random
+import hashlib
+import logging
+import re
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def compute_md5hash(hashstring):
+    """Return the MD5 Hash of a given string"""
+    m = hashlib.md5()
+    m.update(hashstring.encode('utf-8'))
+    return m.hexdigest()
+
+
+def update_label(source, labels):
+    # sort so we do larges matches first removing issues on things like 'ip' used in longer strings
+    labels.sort(key=len, reverse=True)
+    for label in labels:
+        source = source.replace(label + ":", "\"" + label + "\":")
+
+    return source
+
+
+DEFAULT_IP = '192.168.1.254'
+
+
+class BTSmartHub(object):
```

Comment:
Moving everything into a class means that the Home Assistant code will need to be updated. https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/bt_smarthub/device_tracker.py#L83-L84

Reply:
Will you give this a go?
[https://github.com/jxwolstenholme/home-assistant/blob/dev/homeassistant/components/bt_smarthub/device_tracker.py](https://github.com/jxwolstenholme/home-assistant/blob/dev/homeassistant/components/bt_smarthub/device_tracker.py)

Reply:
I can try and test this over the next few days.